pkg/controller/main-controller.go

```diff
@@ -813,6 +813,10 @@ func (c *Controller) syncHandler(key string) (Result, error) {
 			c.recorder.Event(tenant, corev1.EventTypeWarning, "MissingCreds", "Tenant is missing root credentials")
 			return WrapResult(Result{}, nil)
 		}
+		if k8serrors.IsNotFound(err) {
```

Comment:
Would it keep logging events again and again?

Reply:
It's fine to keep logging this. It shouldn't happen during normal operation and AFAIK this is common in Kubernetes. It is often listed as "event occurred (6x)", so I think it will just keep track of the first/last occurence and the number of occurrences.